test/unit/rules/lint-required-context-role-test.js

```diff
@@ -0,0 +1,204 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+
+const ROOTS = {
+  row: ['rowheader', 'columnheader', 'gridcell'],
+  group: ['listitem', 'menuitem', 'menuitemradio', 'treeitem'],
+  menu: ['menuitem', 'menuitemcheckbox', 'menuitemradio'],
+  menubar: ['menuitem', 'menuitemcheckbox', 'menuitemradio'],
+  list: ['listitem'],
+  listbox: ['option'],
+  grid: ['row', 'rowgroup'],
+  rowgroup: ['row'],
+  treegrid: ['row'],
+  tablist: ['tab'],
+  tree: ['treeitem'],
+};
+
+const VALID_PARENTS_FOR_CHILD = Object.keys(ROOTS).reduce((result, keyName) => {
+  const childs = ROOTS[keyName];
+  childs.forEach(childName => {
+    if (!(childName in result)) {
+      result[childName] = [];
+    }
+    result[childName].push(keyName);
+  });
+  return result;
+}, {});
+Object.keys(VALID_PARENTS_FOR_CHILD).forEach(key => {
+  VALID_PARENTS_FOR_CHILD[key].sort();
+});
+const errorMessage = role => {
+  return `You have an element with the role of "${role}" but it is missing the required (immediate) parent element of "[${VALID_PARENTS_FOR_CHILD[
+    role
+  ].join(', ')}]". Reference: https://www.w3.org/TR/wai-aria-1.0/roles#${role}.`;
+};
```

Comment:
we should `require()` the original function instead of duplicating the logic here.